src/components/application_manager/src/policies/policy_handler.cc

```diff
@@ -1501,12 +1501,30 @@ void PolicyHandler::OnSnapshotCreated(const BinaryMessage& pt_string) {
     return;
   }
 
-  AppIdURL app_url = policy_manager_->GetNextUpdateUrl(urls);
-  while (!IsUrlAppIdValid(app_url.first, urls)) {
-    app_url = policy_manager_->GetNextUpdateUrl(urls);
+  const uint32_t app_id = GetAppIdForSending();
+  const std::string app_policy_id =
+      application_manager_.application(app_id)->policy_app_id();
+
+  EndpointUrls::const_iterator app_item = std::find_if(
+      urls.begin(),
+      urls.end(),
+      [app_policy_id](const EndpointData& ed) {
+
+        return helpers::Compare<std::string, helpers::EQ, helpers::ONE>(
+            ed.app_policy_id, policy::kDefaultId, app_policy_id);
+      });
+
+  if (app_item->url.empty()) {
```

Comment:
@ZhdanovP Crash if no app item was found...

Reply:
@AByzhynar done according for all previous comments